### PR TITLE
Fix/ Add NotFound test to WaitForInclusion

### PR DESCRIPTION
**Pull request by gdbelvin**
`client.WaitForInclusion` was erroring when `TreeSize == 0` and several other edge cases. 

- Added test
- Simplified logic

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -224,36 +224,22 @@ func (c *LogClient) WaitForInclusion(ctx context.Context, data []byte) error {
 	}
 
 	// Fetch the current Root to improve our chances at a valid inclusion proof.
-	root, err := c.UpdateRoot(ctx)
+	// It is illegal to ask for an inclusion proof with TreeSize = 0.
```
Comment (Martin2112) [inline client/client.go]: Not so much that it's illegal, it doesn't make sense because the tree has no entries.

Comment (gdbelvin) [reply client/client.go]: Well, if one does try to do that, the server will respond with an error :-)
